src/func.d

```diff
@@ -4486,7 +4486,7 @@ public:
         return this;
     }
 
-    override const(char)* kind()
+    override const(char)* kind() const
     {
         // GCC requires the (char*) casts
         return (tok != TOKfunction) ? cast(char*)"delegate" : cast(char*)"function";
```

Comment:
Also, a major goal of mine is to improve encapsulation in dmd. const-correctness is necessary to achieve that.